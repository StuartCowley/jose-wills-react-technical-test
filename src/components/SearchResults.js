import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ searchResults }) => {
  if (!searchResults.length) {
    return (
      <p>
        Have you searched yet? Either there's no results, or start exploring!
      </p>
    );
  }

  return (
    <>
      <p>Search Results</p>
      <div className="container">
        {searchResults.map((result, index) => (
          <img src={result} alt={`Result ${index}`} key={result} />
        ))}
      </div>
    </>
  );
};

export default SearchResults;
