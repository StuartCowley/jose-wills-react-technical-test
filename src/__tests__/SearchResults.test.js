import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  const validProps = [
    "../images/testImage1.jpeg",
    "../images/testImage2.jpeg",
    "../images/testImage3.jpeg",
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults searchResults={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the right number of images", () => {
    render(<SearchResults searchResults={validProps} />);

    const images = screen.getAllByRole("img");

    expect(images).toHaveLength(3);
  });

  it("renders 'Click and hold the images to zoom in!' after searching", () => {
    render(<SearchResults searchResults={validProps} />);

    const text = screen.getByText("Click and hold the images to zoom in!");

    expect(text).toBeInTheDocument();
  });

  it("renders message if no results or no search", () => {
    render(<SearchResults searchResults={[]} />);

    const text = screen.getByText(
      "Have you searched yet? Either there's no results, or start exploring!"
    );

    expect(text).toBeInTheDocument();
  });
});
