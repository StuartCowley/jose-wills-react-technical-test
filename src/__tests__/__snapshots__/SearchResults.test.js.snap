// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <p>
    Click and hold the images to zoom in!
  </p>
  <div
    class="container"
  >
    <img
      alt="Result 0"
      class="search-result__image"
      src="../images/testImage1.jpeg"
    />
    <img
      alt="Result 1"
      class="search-result__image"
      src="../images/testImage2.jpeg"
    />
    <img
      alt="Result 2"
      class="search-result__image"
      src="../images/testImage3.jpeg"
    />
  </div>
</DocumentFragment>
`;
